src/pages/plans/Plans.tsx

```diff
@@ -1,9 +1,111 @@
 import "./Plans.css";
+import { useState } from "react";
+import {
+  useGetUserPricesQuery,
+  useUpdatePriceMutation,
+} from "../../app/services/PriceService";
+import toast from "react-hot-toast";
+import useAuthentication from "../../hooks/useAuthentication";
+import { formatCurrency } from "../../utils/Formatter";
 
 function Plans() {
+  const { userId } = useAuthentication();
+  const { data: prices, refetch } = useGetUserPricesQuery(userId!);
+  const [newAmount, setNewAmount] = useState<number | "">("");
+  const [selectedPriceId, setSelectedPriceId] = useState("");
+  const [updatePrice, { isLoading: isUpdating }] = useUpdatePriceMutation();
+  const selectedPrice = prices?.find((p) => p.id === selectedPriceId);
+
+  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const rawValue = e.target.value.replace(/\D/g, "");
+    if (rawValue === "") {
+      setNewAmount("");
+    } else {
+      setNewAmount(Number(rawValue) / 100);
+    }
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!selectedPriceId) {
+      toast.error("Por favor, selecciona un plan.");
+      return;
+    }
+
+    if (!newAmount || isNaN(Number(newAmount))) {
+      toast.error("Ingresa un monto válido.");
+      return;
+    }
+
+    updatePrice({
+      priceId: selectedPriceId,
+      amount: Number(newAmount),
+    })
+      .unwrap()
+      .then(() => {
+        toast.success("Precio actualizado con éxito");
+        setNewAmount("");
+        refetch();
+      })
+      .catch((err) => {
+        toast.error("Error al actualizar el precio");
+        console.error(err);
+      });
+  };
+
   return (
-    <div>
-      <p>Plans</p>
+    <div className="plansContainer">
+      <div className="selectContainer">
+        <label className="label" htmlFor="planSelect">
+          Seleccionar un plan o frecuencia
+        </label>
+        <select
+          id="planSelect"
+          className="select"
+          value={selectedPriceId}
+          onChange={(e) => setSelectedPriceId(e.target.value)}
+        >
+          <option value="">- Selecciona una opción -</option>
+          {prices?.map((price) => (
+            <option key={price.id} value={price.id}>
+              {price.name}
+            </option>
+          ))}
+        </select>
+      </div>
+
+      {selectedPrice && (
+        <form className="amountsContainer" onSubmit={handleSubmit}>
+          <div className="actualAmount">
+            <label className="label">Monto Actual</label>
+            <input
+              className="input"
+              readOnly
+              value={formatCurrency(selectedPrice.amount)}
+            />
+          </div>
+          <div className="newAmount">
+            <label className="label">Monto Nuevo</label>
+            <input
+              className="input"
+              type="text"
+              value={newAmount === "" ? "" : formatCurrency(newAmount)}
```

Comment:
Y esto no te haria falta


```suggestion
              value={formatCurrency(newAmount)}
```